packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart/ForceFunctionContainerStartExecutor.ts

```diff
@@ -0,0 +1,164 @@
+  // The function binary. It must be used as a single instance. Multiple class to the getFunctionBinary
+  // function will not provide the same instance of the binary.
+  private functionsBinary: Benny | undefined;
+
+  public async setupFunctionListeners(
+    functionDirPath: string,
+    functionDisposable: vscode.Disposable
+  ): Promise<void> {
+    this.functionsBinary = await getFunctionsBinary();
+
+    const writeMsg = (msg: { text: string; timestamp: string }) => {
+      const outputMsg = msg.text;
+
+      if (outputMsg) {
+        channelService.appendLine(outputMsg);
+
+        const matches = String(outputMsg).match(
+          FUNCTION_RUNTIME_DETECTION_PATTERN
+        );
+        if (matches && matches.length > 1) {
+          FunctionService.instance.updateFunction(functionDirPath, matches[1]);
+        }
+      }
+    };
+
+    const handleError = (error: any) => {
+      functionDisposable.dispose();
+      let unexpectedError = true;
+      (Object.keys(
+        forceFunctionStartErrorInfo
+      ) as ForceFunctionStartErrorType[]).forEach(errorType => {
+        const {
+          cliMessage,
+          errorNotificationMessage
+        } = forceFunctionStartErrorInfo[errorType];
+        if (error.text?.includes(cliMessage)) {
+          unexpectedError = false;
+          telemetryService.sendException(errorType, errorNotificationMessage);
+          notificationService.showErrorMessage(errorNotificationMessage);
+          channelService.appendLine(errorNotificationMessage);
+        }
+      });
+
+      if (unexpectedError) {
+        const errorNotificationMessage = nls.localize(
+          this.UNEXPECTED_ERROR_KEY
+        );
+        telemetryService.sendException(
+          this.UNEXPECTED_ERROR_KEY,
+          errorNotificationMessage
+        );
+        notificationService.showErrorMessage(errorNotificationMessage);
+        channelService.appendLine(errorNotificationMessage);
+      }
+      channelService.showChannelOutput();
+    };
+
+    this.functionsBinary.on(BINARY_EVENT_ENUM.PACK, writeMsg);
+    this.functionsBinary.on(BINARY_EVENT_ENUM.CONTAINER, writeMsg);
+
+    this.functionsBinary.on(BINARY_EVENT_ENUM.LOG, (msg: any) => {
+      if (msg.level === 'debug') return;
+      if (msg.level === 'error') {
+        handleError(msg);
+      }
+
+      if (msg.text) {
+        writeMsg(msg);
+      }
+
+      // evergreen:benny:message {"type":"log","timestamp":"2021-05-10T10:00:27.953248-05:00","level":"info","fields":{"debugPort":"9229","localImageName":"jvm-fn-init","network":"","port":"8080"}} +21ms
+      if (msg.fields && msg.fields.localImageName) {
+        channelService.appendLine(`'Running on port' :${msg.fields.port}`);
+        channelService.appendLine(
+          `'Debugger running on port' :${msg.fields.debugPort}`
+        );
+      }
+    });
+    // Allows for showing custom notifications
+    // and sending custom telemtry data for predefined errors
```

Comment:
typo: telemetry

Reply:
Truth be told, I'm a terrible speller.  I use a VS Code extension named "Code Spell Checker" and it does a good job of finding spelling errors.